coupled_biased_random_walks/detect.py

```diff
@@ -91,13 +91,40 @@ def score(self, observation_iterable):
 
     def _score(self, observation):
         """
-        Computes the weighted anomaly score for an observation
+        Computes the weighted anomaly score (object_score in the paper) for an observation
         :param observation: dict of the form {feature_name: feature_value, ...}
         """
         return sum(self._get_feature_relevance(item) * \
                    self._stationary_prob.get(item, self._unknown_feature_score)
                     for item in iteritems(observation))
 
+    def value_scores(self, observation_iterable):
+        """
+        Compute an anomaly sub-score for each value of the observation in observation_iterable
+        :param observation_iterable: iterable of dict observations with each dict
+        of the form {feature_name: feature_value, ...}
+        Return dict with sub score of each value of the observation/object of the form:
+        {feature_name: weighted score of value of feature, ...}
+        (sub-scores sum up to score(self, observation_iterable))
+        """
+        if not (self._feature_relevance and self._stationary_prob):
+            raise CBRWScoreError()
+        if isinstance(observation_iterable, dict):
+            observation_iterable = [observation_iterable]
+        return np.array([self._value_scores(obs) for obs in observation_iterable])
+
+    def _value_scores(self, observation):
+        """
+        Computes the weighted value scores for each feature value of an observation
+        """
+        score_keys = []
+        value_scores = []
+        for item in iteritems(observation):
+            score_keys.append(item[0])
```

Comment:
For consistency with other parts of the code, we should probably use the helper function `get_feature_name` rather than index directly into the tuple:
```python
score_keys.append(get_feature_name(item))
```